COMPILING.md

```diff
@@ -11,6 +11,11 @@ developers don't have C compilers installed and don't need to run or modify the
 codegen, the build can skip it. To skip, create the file
 `<project-root>/gradle.properties` and add `skipCodegen=true`.
 
+Some parts of grpc-java depend on Android. Since many Java developers don't have
+the Android SDK installed and don't need to run or modify the Android
```

Comment:
Are there instructions to install the Android SDK in case someone doesn't want to "skipAndroid"?

Reply:
It is basically download the SDK and set $ANDROID_HOME. But there's some complications that make it hard to be concise. I think I'll let @dapengzhang0 or @voidzcy add a documentation section later in this document, but after this is merged.

(Among the problems: ANDROID_HOME is actually deprecated now in favor of ANDROID_SDK_HOME, but we'd need to upgrade to a newer gradle plugin for ANDROID_SDK _HOME to work)

Reply:
Sure. It's in #6593.